sdk/storage/azure-storage-blob-nio/src/main/java/com/azure/storage/blob/nio/AzureFileSystemProvider.java

```diff
@@ -1159,4 +1166,21 @@ private String extractAccountEndpoint(URI uri) {
 
         return endpoint;
     }
+
+    /**
+     * Sets the default configurations for this {@code FileSystemProvider}. The configurations will be applied if a call
+     * to {@link Paths#get(URI)} is made and there is not an open file system pointing to the specified endpoint. In
+     * this case, a corresponding file system will be opened. Otherwise, the already opened file system will be used.
+     * The endpoint must be specified in the uri.
+     *
+     * A file system will only be opened automatically if these defaults are set. If they are not set, attempting to
+     * retrieve a path in an unopened file system will continue to throw a {@link FileSystemNotFoundException}.
+     *
+     * @param config The configurations map. Please see the docs on {@link AzureFileSystemProvider for more information}
+     */
+    public static void setDefaultConfigurations(Map<String, Object> config) {
+        defaultConfigurations = Collections.unmodifiableMap(config);
```

Comment:
this doesn't block somebody with reference to `config` from making further modifications. Should map be copied?

Reply:
Updated.